extend/generic-writer/configuration/configuration.md

```diff
@@ -0,0 +1,718 @@
+
+##### Example
+
+Let's have this table on the input:
+
+| id | date       | name  | email      | address |
+|----|------------|-------|------------|---------|
+| 1  | 01.01.2020 | David | d@test.com | asd     |
+| 2  | 01.02.2020 | Tom   | t@test.com | asd     |
+
+Consider following request options:
+
+```json
+"request_options": {
+    "api_request": {
+        "method": "POST",
+        "endpoint_path": "/user/[[id]]",
+        "query_parameters": {
+            "date": {
+                    "attr": "date"
+                    }
+        }
+    },
+    "content": {
+        "content_type": "JSON",
+        "iterate_by_columns": [
+        "id", "date"
+        ]
+    }
+}
+```
+
+The writer will run in two iterations:
+
+**FIRST** With data
+
+| name  | email      | address |
+|-------|------------|---------|
+| David | d@test.com | asd     |
+
+Sent to `www.example.com/api/user/1?date=01.01.2020`
+
+
+
+
+**SECOND** with data
+
+| name  | email      | address |
+|-------|------------|---------|
+| Tom   | t@test.com | asd     |
+
+Sent to `www.example.com/api/user/2?date=01.02.2020`
+
+## Dynamic Functions
+
+The application support functions that may be applied on parameters in the configuration to get dynamic values.
+
+Currently these functions work only in the `user_parameters` scope. 
+Place the required function object instead of the user parameter value.
+
+The function values may refer to another user params using `{"attr": "custom_par"}`
+
+**NOTE:** If you are missing any function let us know or place a PR to our [repository](https://bitbucket.org/kds_consulting_team/kds-team.wr-generic/src/).
+It's as simple as adding an arbitrary method into the [UserFunctions class](https://bitbucket.org/kds_consulting_team/kds-team.wr-generic/src/master/src/user_functions.py#lines-7)
+
+**Function object**
+    
+```json
+{ "function": "string_to_date",
+                "args": [
+                  "yesterday",
+                  "%Y-%m-%d"
+                ]
+              }
+```
+
+#### Function Nesting
+
+Nesting of functions is supported:
+
+```json
+{
+   "user_parameters":{
+      "url":{ 
+         "function":"concat",
+         "args":[
+            "http://example.com",
+            "/test?date=",
+            { "function": "string_to_date",
+                "args": [
+                  "yesterday",
+                  "%Y-%m-%d"
+                ]
+              }
+         ]
+      }
+   }
+}
+
+```
+
+#### string_to_date
+
+Function converting string value into a datestring in specified format. The value may be either date in `YYYY-MM-DD` format,
+or a relative period e.g. `5 hours ago`, `yesterday`,`3 days ago`, `4 months ago`, `2 years ago`, `today`.
+
+The result is returned as a date string in the specified format, by default `%Y-%m-%d`
+
+The function takes two arguments:
+
+1. [REQ] Date string
+2. [OPT] result date format. The format should be defined as in http://strftime.org/
```

Comment:
linkk, [OPTIONAL] if you want to keep it consistent (but i'd omit that) and why is the format even optional? there must be some default then